packages/data-explorer/src/repository/dataRepository.ts

```diff
@@ -113,32 +113,41 @@ const getTableDataDeepReference = async (
   }
 
   const expandReferencesQuery = buildExpandedAttributesQuery(metaData, coloms)
-  const size = _buildSizeQueryParam(dataDisplayLimit)
-  const request = addFilterIfSet(`/api/data/${tableId}?${size}${expandReferencesQuery}`, rsqlQuery)
+  let pageQuery = ''
+  if (pagination) {
+    pageQuery = `${buildQueryString(pagination)}&`
+  }
+  const request = addFilterIfSet(`/api/data/${tableId}?${pageQuery}${expandReferencesQuery}`, rsqlQuery)
   const response = (await client.get<DataApiResponse>(request)).data
-  const result = { items: response.items.map((item: DataApiResponseItem) => levelNRowMapper(item)) }
-  return result
+
+  const items = response.items.map((item: DataApiResponseItem) => levelNRowMapper(item))
+  return { items, response }
 }
 
-const getTableDataWithLabel = async (tableId: string, metaData: MetaData, columns: string[], rsqlQuery?: string, dataDisplayLimit?: Number) => {
+const getTableDataWithLabel = async (tableId: string, metaData: MetaData, columns: string[], rsqlQuery?: string, pagination?: Object) => {
```

Comment:
would suggest to type pagination object

Reply:
Yes, see https://github.com/molgenis/molgenis-frontend/pull/477